src/common/utils/utils.go

```diff
@@ -277,3 +281,20 @@ func FindNamedMatches(regex *regexp.Regexp, str string) map[string]string {
 	}
 	return results
 }
+
+// NextSchedule return next scheduled time with a cron string and current time provided
+// the cron string could contain either 5 tokens or 6 tokens,
+// if the cron string contains 6 tokens, it converts the cron string to 5 tokens by removing the seconds field
+// if the cron string is invalid, it returns a zero time
+func NextSchedule(cron string, curTime time.Time) time.Time {
+	cr := strings.TrimSpace(cron)
+	if _, err := cronlib.Parse(cron); err != nil {
+		return time.Time{}
+	}
+	parts := strings.Split(cr, " ")
+	// remove the seconds field, it should always be 0 in harbor cron string
+	if len(parts) > 5 {
+		cr = cr[2:]
+	}
+	return cronexpr.MustParse(cr).Next(curTime)
```

Comment:
This function uses two library to parse and compute next time, but I confuse here why not just use `robfig/cron.Parse(cron).Next(curTime)`.

Reply:
Updated